iOS/Flows/People/PeopleCoordinator.swift

```diff
@@ -36,9 +43,230 @@ class PeopleCoordinator: PresentableCoordinator<[Person]> {
             self.peopleNavController.prepareForInvitations()
             
             Task {
-                await self.updateInvitation()
+                await self.inviteRemainingPeople()
             }.add(to: self.taskPool)
         }
     }
 }
 
+extension PeopleCoordinator {
+
+    /// Invites the next person in the array of people to invite and keeps doing so until there are no more people to invite.
+    /// Once there are no more people to invite, the finishInviting phase is called.
+    @MainActor
+    func inviteRemainingPeople() async {
+        // If there are no more people to invite, then finish the flow.
+        guard var person = self.peopleToInvite[safe: self.inviteIndex] else {
+            await self.finishInviting()
+            return
+        }
+
+        guard !Task.isCancelled else { return }
+
+        self.inviteIndex += 1
+
+        // If the invited person is already connected, there's no need to formally invite them.
+        // Move on to the next person
+        if let connection = try? await person.connection?.retrieveDataIfNeeded(),
+            connection.status == .accepted {
+            // Do nothing because the person is already connected.
+        } else {
+            // The person is not connected, we'll need to try inviting them to connect.
+            if let connection = await self.invite(person: person) {
+                // If a connection was made, update the person's connection info.
+                person.connection = connection
+            }
+        }
+
+        // Keep track of who was connected/invited.
+        self.invitedPeople.append(person)
```

Comment:
Because everything is async, we can append people to the invite list in one spot.